eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -96,8 +96,9 @@ object OnionMessages {
     } else {
       payloadSize.toInt
     }
-    val Sphinx.PacketAndSecrets(packet, _) = Sphinx.create(sessionKey, packetSize, route.blindedNodes.map(_.blindedPublicKey), payloads, None)
-    (route.introductionNodeId, OnionMessage(route.blindingKey, packet))
+    Sphinx.create(sessionKey, packetSize, route.blindedNodes.map(_.blindedPublicKey), payloads, None).map {
+      case Sphinx.PacketAndSecrets(packet, _) => (route.introductionNodeId, OnionMessage(route.blindingKey, packet))
+    }
```

Comment:
https://github.com/ACINQ/eclair/pull/2087/commits/4c9b77c1045ef437deb40f9c4c869a3065a93dc1 changed this to a `.get` without even a `Try`.

Reply:
You've only changed it here where the `Try` is no longer needed, my comment was more general. It's about all the places where you've added `.map( x =>` after the calls that return a `Try[X]`, I think the code is cleaner if you just add `.get` and wrap the whole thing in `Try{}`. But if you disagree you can ignore it.

Reply:
I'd rather keep a functional programming pattern (`.map`) instead of switching to an imperative programming pattern.
On top of that, doing `.get` means we're actually throwing an exception when it fails, which means capturing a stack trace (somewhat expensive if done in a tight loop) just to catch it again instantly, which is wasteful, so my rule of thumb is to usually avoid it when we can (even though in that case I'm not sure we would unwind the stack trace, maybe it doesn't change much, but better stay away from exceptions for flow control)...